### ISSUE TITLE
Overfitting?

### ISSUE THREAD

**Issue by mazatov** [VisualComputingInstitute/triplet-reid#93]
This is more of a discussion than an issue. I was wondering about potential overfitting in Trinet and how one could estimate it. 

Normally I always have a validation dataset and I can compare accuracy/loss between validation dataset and training dataset to see if I'm overfitting. However, here we train for 25000 epochs and there don't seem to be any validation checks as far as I understand. So I'm wondering how do we know that the model is not overfitted on the training dataset? 

I'm training the model on a smaller dataset and clearly observe overfitting where the model performs much better on the training set and than the test set. So I'm trying to think of strategies to avoid it.
